Terminal.Gui/View/Layout/ViewLayout.cs

```diff
@@ -499,21 +504,15 @@ public Pos Y
     public event EventHandler Initialized;
 
     /// <summary>Converts a <see cref="Bounds"/>-relative rectangle to a screen-relative rectangle.</summary>
-    public virtual Rectangle BoundsToScreen (Rectangle bounds)
+    public Rectangle BoundsToScreen (Rectangle bounds)
     {
+        // Translate bounds to Frame (our SuperView's Bounds-relative coordinates)
         Point boundsOffset = GetBoundsOffset ();
-        bounds.Offset(Frame.X + boundsOffset.X, Frame.Y + boundsOffset.Y);
 
-        View super = SuperView;
+        Rectangle screen = FrameToScreen ();
+        screen.Offset (boundsOffset.X + bounds.X, boundsOffset.Y + bounds.Y);
```

Comment:
Use of Point.Add would make this more efficient.

Reply:
Help me out here. What would that look like?

Reply:
Naive case is taking your boundsOffset and bounds variables and doing something like:

screen.Offset(boundsOffset.Add(bounds));

And then whatever other cleanup, simplification, or optimization you might see from there.

Basically, avoiding explicit math on the individual properties/fields of Point, Size, and Rectangle is a good thing, whenever possible.